noSelectStar/testdata/src/a/a.go

```diff
@@ -0,0 +1,37 @@
+package a
+
+func Select(args ...string)  {}
+func Columns(args ...string) {}
+
+func example() {
+	// These should trigger warnings
+	_ = "SELECT * FROM users" // want `do not use SELECT \*: explicitly select the needed columns instead`
+	_ = "select * from table" // want `do not use SELECT \*: explicitly select the needed columns instead`
+
+	// These should not trigger warnings
+	_ = "SELECT id, name FROM users"
+	_ = "Just a * by itself"
+	_ = "SELECT"
+
+	// These should trigger warnings for Select function
+	Select("*")     // want `do not use Select with \*: explicitly select the needed columns instead`
+	Select("id, *") // want `do not use Select with \*: explicitly select the needed columns instead`
+	Select("id", "*", "name") // want `do not use Select with \*: explicitly select the needed columns instead`
+	Select("id", "name", "*") // want `do not use Select with \*: explicitly select the needed columns instead`
+
+	// These should not trigger warnings for Select function
+	Select("id, name")
```

Comment:
Can we also add a test case like:

```
query := s.getQueryBuilder().
		Select("*").
		From("Channels").
		Where(sq.Eq{"Id": ids})
```

to test for completeness?

Reply:
Good catch: the previous code was entirely incompatible with our actual pattern here. I confess to offloading too much to aider here: I suppose this is analogous to the self-driving car problem where the drivers risk not paying sufficient attention to take over when things don't go right.

I still used aider to fix this, but enhanced the test cases manually.

Reply:
Good that we have humans to review code :joy: